classes/image_types_tegra.bbclass

```diff
@@ -72,6 +72,14 @@ python() {
     valid_types = d.getVar('TEGRAFLASH_PACKAGE_FORMATS').split()
     if pt not in valid_types:
         bb.fatal("TEGRAFLASH_PACKAGE_FORMAT %s not recognized, supported types are: %s" % (pt, ', '.join(valid_types)))
+
+    # handle external provider for dtb (adapted from kernel-fitimage.bbclass)
+    providerdtb = d.getVar("PREFERRED_PROVIDER_virtual/dtb")
+    if providerdtb:
+        d.appendVar('CONVERSION_DEPENDS_bup-payload', ' virtual/dtb:do_populate_sysroot')
+        d.appendVarFlag('do_image_tegraflash', 'depends', ' virtual/dtb:do_populate_sysroot')
+        # d.appendVarFlag('create_bup_payload_image', 'depends', ' virtual/dtb:do_populate_sysroot')
+        d.setVar('EXTERNAL_KERNEL_DEVICETREE', "${RECIPE_SYSROOT}/boot/devicetree")
```

Comment:
Yes, I can do it this way. I chose the python function since it was this way in the kernel/fitimage class, but I see your points, and furthermore it is more consistent with other schemes used here.